infra/resources/modules/commons/function_async.tf

```diff
@@ -131,6 +131,8 @@ module "subscription_async_fn" {
   }
   subnet_id = module.subscription_async_snet.id
 
+  allowed_subnets = [module.subscription_async_snet.id]
```

Comment:
I suggest to add a private endpoint rather than filtering traffic on subnets

P.S. I thought private endpoints were already enabled, this might explain why APIM can connect to this resource

Reply:
If I'm not wrong set subnet is the only way to make this resource not available publicly

Reply:
Nope, app service and function apps vnet integration happens only for outbound connectivity.

Instead, for inbound ones you can setup the firewall by allowing a set of IPs (what you're actually doing) sticking to the public network. Otherwise, you can close the public network connectivity and creates a private endpoint to only accept incoming connections from the vnet where the private endpoint is and using the Microsoft network backbone (more performant and secure). The latter is the preferred option.

As I'm writing, I'm realizing the module you're using doesn't allow to block public connectivity as doesn't support the property [public_network_access_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_function_app#public_network_access_enabled) 🤦🏻 

I might open a PR

Reply:
Exactly the module we are using now doesn't support the property `public_network_access_enabled`, sorry for not being clear earlier. Right now, it seems like subnet routing is our only choice with the module we're using. We're thinking about eventually moving over to the [function_app DX module](https://github.com/pagopa/dx/tree/main/infra/modules/azure_function_app), but we're planning to tackle that as a separate task

Reply:
We need the same property elsewhere as well, so we're adding it: https://github.com/pagopa/terraform-azurerm-v3/pull/315

Reply:
Done ✅